pylops/utils/seismicevents.py

```diff
@@ -397,6 +398,104 @@ def linear3d(
     return d, dwav
 
 
+def parabolic3d(
+    x: npt.NDArray,
+    y: npt.NDArray,
+    t: npt.NDArray,
+    t0: Union[float, Tuple[float]],
+    px: Union[float, Tuple[float]],
+    py: Union[float, Tuple[float]],
+    pxx: Union[float, Tuple[float]],
+    pyy: Union[float, Tuple[float]],
+    amp: Union[float, Tuple[float]],
+    wav: npt.NDArray,
+) -> Tuple[npt.NDArray, npt.NDArray]:
+    r"""Parabolic 3D events
+
+    Create 3d parabolic events given intercept time,
+    slowness, curvature, and amplitude of each event.
+
+    Parameters
+    ----------
+    x : :obj:`numpy.ndarray`
+        space axis in x direction
+    y : :obj:`numpy.ndarray`
+        space axis in y direction
+    t : :obj:`numpy.ndarray`
+        time axis
+    t0 : :obj:`tuple` or :obj:`float`
+        intercept time at :math:`x=0` and :math:`y=0` of each parabolic event
+    px : :obj:`tuple` or :obj:`float`
+        slowness of each parabolic event in x direction
+    py : :obj:`tuple` or :obj:`float`
+        slowness of each parabolic event in y direction
+    pxx : :obj:`tuple` or :obj:`float`
+        curvature of each parabolic event
+    amp : :obj:`tuple` or :obj:`float`
+        amplitude of each linear event
+    wav : :obj:`numpy.ndarray`
+        wavelet to be applied to data
+
+    Returns
+    -------
+    d : :obj:`numpy.ndarray`
+        data without wavelet of size
+        :math:`[n_y \times n_x \times n_t]`
+    dwav : :obj:`numpy.ndarray`
+        data with wavelet of size
+        :math:`[n_y \times n_x \times n_t]`
+
+    Notes
+    -----
+    Each event is created using the following relation:
+
+    .. math::
+        t_i(x, y) = t_{0,i} + p_{x,i} x + p_{y,i} x + p_{xx,i} x^2 + p_{yy,i} y^2
+
+    """
+    if isinstance(t0, (float, int)):
+        t0 = (t0,)
+    if isinstance(px, (float, int)):
+        px = (px,)
+    if isinstance(py, (float, int)):
+        py = (py,)
+    if isinstance(pxx, (float, int)):
+        pxx = (pxx,)
+    if isinstance(pyy, (float, int)):
+        pyy = (pyy,)
```

Comment:
Use the function `pylops._internal._value_or_sized_to_array`. Avoid isinstance!

Reply:
This is likely a remnant of the past that was never fixed and when I made the `parabolic3d` I just copy pasted it from the `linear3d`... I have not removed all `isinstance` occurencies in this file